### PR TITLE
Ss rf handle user stopping location sharing

### PR DESCRIPTION
**Pull request by snsebro**
## Description

Updated geolocation getPosition function to watchPosition, allowing for a user's location to be tracked when allowed as opposed to called once.

Added clearWatch function to allow a user to stop sharing location, at which point the app returns to its default state.

Updated loading component with icon in nav as opposed message at top of page.

## Related Issue

Jira Issue: https://the-collab-lab.atlassian.net/browse/TCL20-22?atlOrigin=eyJpIjoiOWZmZTg3ZDM2YzhhNDBmOGFlYWNmOTVhY2E4NmI4MzciLCJwIjoiaiJ9 
## Acceptance Criteria

- [x] When the user has shared their location, they should have available a prompt (e.g., a button or link) they can tap or click to stop sharing

**When the user taps/clicks the stop sharing prompt, we…**

- [x] unregister the handler that is watching their position

- [x] remove the marker showing their location on the map

- [x] return the location sharing prompt to its original state to allow them to start sharing again

- [x] Given a user who is sharing their device location with the app, when they tap/click the “stop location sharing” prompt, then the app should stop watching their location and return to the default state

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓   | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓   | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
image

### After
image

image

### CODE REVIEW

File: src/components/LoadingMessage.css [hunk 1]
```diff
@@ -1,15 +1,31 @@
 .loading-message {
-  height: 50px;
-  width: 100%;
-  border: 1px solid black;
   z-index: 1;
-  display: flex;
-  justify-content: center;
-  align-items: center;
+  display: inline;
+  /* justify-content: center;
```
Comment (segdeha) [inline src/components/LoadingMessage.css]: Looks like we can remove this commented code, yah?

File: src/components/LoadingMessage.css [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+  animation: spin 1s ease-in-out infinite;
+  -webkit-animation: spin 1s ease-in-out infinite;
+}
+
+@keyframes spin {
+  to {
+    -webkit-transform: rotate(360deg);
+  }
+}
+@-webkit-keyframes spin {
```
Comment (segdeha) [inline src/components/LoadingMessage.css]: I wonder how old of a device you need to have to need this fallback? Could probably get away with removing it? 🤔

File: src/components/Nav.css [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
   background-color: white;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
 }
 
 ul {
   list-style: none;
   padding: 0;
+  width: 100%;
```
Comment (segdeha) [inline src/components/Nav.css]: The default `width` for block-level elements is `100%`. Was this not working as-expected without this here?

File: src/context/LocationsContext.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ const LocationsContextProvider = (props) => {
   //state is updated in the Map component
   const [newCenter, setNewCenter] = useState();
   const [loading, setLoading] = useState({ loading: false, message: '' });
+  const [id, setId] = useState();
```
Comment (segdeha) [inline src/context/LocationsContext.js]: `id` is a pretty generic variable name. Could we rename this to `watchId` perhaps?

File: src/context/LocationsContext.js [hunk 5]
```diff
@@ -29,23 +30,34 @@ const LocationsContextProvider = (props) => {
...
-          setLoading({ loading: false, message: '' });
-        },
-        (error) => {
-          console.error(error);
-          setLoading({
-            loading: false,
-            message: 'location services turned off',
-          });
-        },
+      setId((id) =>
```
Comment (segdeha) [inline src/context/LocationsContext.js]: I'm a little confused by this function. We're passing in `id` but that variable is available from the enclosing scope, so you could just omit that. But more importantly, it doesn't appear to set the ID, or am I missing something?

This looks more like the function should be named `enableWatch` or something, if I'm understand its purpose?

File: src/context/LocationsContext.js [hunk 6]
```diff
@@ -29,23 +30,34 @@ const LocationsContextProvider = (props) => {
...
+            });
+          },
+        ),
+      );
+    }
+    if (navigator.geolocation && !allowLocation) {
+      navigator.geolocation.clearWatch(id);
+      setUserLocation(undefined);
+      getLocations(
+        defaultCoordinates.lat,
```
Comment (segdeha) [inline src/context/LocationsContext.js]: Ah, OK. I looked at the ticket again and I see that I said “return to the default state”, which is probably why y’all used the default coordinates here. What I think would be a better experience is just leaving the map centered wherever it currently is. I can’t believe you didn’t read my mind! 😅 

I’ll think of a better way to phrase that for future teams!

File: src/pages/Map.js [hunk 7]
```diff
@@ -54,6 +51,7 @@ export default function Map() {
         onDragEnd={(e) => handleNewCenter(e)}
         onGoogleApiLoaded={({ map, maps }) => handleApiLoaded(map, maps)}
       >
+        {/* TODO: Check that this re-renders when user location is updated */}
```
Comment (segdeha) [inline src/pages/Map.js:54]: Is this still needed? If not, please remove. Thanks!

**Review by segdeha — APPROVED** [review]
Wow, great work!!! One little thing is that I caused a misunderstanding by saying the app should “return to the default state”. I should have been more specific about the map center staying put when the user turns off location services. I had a couple of questions about naming things (one of the hard problems of computer science!), but things are functioning as-intended so consider those optional. Once you get the map center thing worked out, feel free to merge! 💯

File: src/components/LoadingMessage.css (hunk 1 above)
Comment (rachel-fischoff) [reply src/components/LoadingMessage.css]: yes!

Comment (snsebro) [reply src/components/LoadingMessage.css]: Went through and got rid of comments

File: src/components/Nav.css (hunk 3 above)
Comment (snsebro) [reply src/components/Nav.css]: It works fine without

File: src/context/LocationsContext.js (hunk 5 above)
Comment (snsebro) [reply src/context/LocationsContext.js]: My understanding of the documentation here is that the `watchPosition` method returns an `id` that you can pass to `clearWatch` to reference the correct position to stop watching. So here we are trying to set that `id` to state and then call it again in `clearWatch` on line 50.

The id isn't read on line 33 so I removed that.

File: src/context/LocationsContext.js (hunk 6 above)
Comment (snsebro) [reply src/context/LocationsContext.js]: haha all good, I removed this so now the map is left in its current view and the user location icon is toggled off

**Pull request merged**